### PR TITLE
server/resource: replace score type with resource kind

### PR DESCRIPTION
**Pull request by huachaohuang**
I think the scorers are too complicated, we can calculate the scores very easily.

### CODE REVIEW

**Comment by huachaohuang** [comment]
PTAL @siddontang @overvenus

File: server/balancer.go [hunk 1]
```diff
@@ -45,48 +45,48 @@ func newCapacityBalancer(cfg *BalanceConfig) *capacityBalancer {
...
+func (cb *capacityBalancer) Balance(cluster *clusterInfo) (float64, *balanceOperator, error) {
 	region, source, target := scheduleStorage(cluster, cb.selector)
 	if region == nil {
-		return nil, nil, nil
+		return 0, nil, nil
+	}
+
+	// Check diff score.
+	diff := source.resourceRatio(cb.kind) - target.resourceRatio(cb.kind)
+	if diff < cb.cfg.MaxDiffScoreFraction {
```
Comment (siddontang) [inline server/balancer.go]: not MinDiff here?

Comment (huachaohuang) [inline server/balancer.go]: Yes, some configs are confusing, I will change them later.

File: server/balancer.go [hunk 2]
```diff
@@ -45,48 +45,48 @@ func newCapacityBalancer(cfg *BalanceConfig) *capacityBalancer {
...
 	addPeerOperator := newAddPeerOperator(region.GetId(), newPeer)
 	removePeerOperator := newRemovePeerOperator(region.GetId(), peer)
-	return score, newBalanceOperator(region, balanceOP, addPeerOperator, removePeerOperator), nil
+	return diff, newBalanceOperator(region, balanceOP, addPeerOperator, removePeerOperator), nil
 }
 
 type leaderBalancer struct {
 	cfg      *BalanceConfig
-	st       scoreType
+	kind     ResourceKind
```
Comment (siddontang) [inline server/balancer.go]: can we use leaderKind directly in leaderBalancer?

File: server/cluster.go [hunk 3]
```diff
@@ -454,18 +454,18 @@ func (c *RaftCluster) collectMetrics() {
...
-			minUsedRatio = s.usedRatio()
-		}
-		if maxUsedRatio < s.usedRatio() {
-			maxUsedRatio = s.usedRatio()
-		}
 		if minLeaderRatio > s.leaderRatio() {
 			minLeaderRatio = s.leaderRatio()
 		}
 		if maxLeaderRatio < s.leaderRatio() {
 			maxLeaderRatio = s.leaderRatio()
```
Comment (siddontang) [inline server/cluster.go]: no need to check min == max ?

**Comment by siddontang** [comment]
Rest LGTM

The code looks more clean now, but should improve the test coverage.

PTAL @overvenus

File: server/balancer.go (hunk 2 above)
Comment (huachaohuang) [inline server/balancer.go]: Then we need to type it in different places :(

File: server/cluster.go (hunk 3 above)
Comment (huachaohuang) [inline server/cluster.go]: No, it's OK if min == max, it means we balance very well.

**Pull request merged**